File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        dummy=ListNode(-1)
        curr=dummy
        t1=list1
        t2=list2
        temp=ListNode()
        while t1 or t2:
            if t1==None:
                temp=t2
                curr.next=temp
                curr=curr.next
                t2=t2.next
            elif t2==None:
                temp=t1
                curr.next=temp
                curr=curr.next
                t1=t1.next
            elif t1.val <= t2.val:
                temp=t1
                curr.next=temp
                curr=curr.next
                t1=t1.next
            elif t2.val<t1.val:
                temp=t2
                curr.next=temp
                curr=curr.next
                t2=t2.next
        return dummy.next
                

        

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode head;
        int x;
        ListNode* curr=&head;
        // ListNode* curr1=list1;
        // ListNode* curr2=list2;
        while(list1 && list2){
            if(list1->val <= list2->val ){
                x=list1->val;
                list1=list1->next;
            }
            else{
                x=list2->val;
                list2=list2->next;
            }
            curr->next= new ListNode(x);
            curr=curr->next;
        }
        while(list1){
            x=list1->val;
            curr->next= new ListNode(x);
            curr=curr->next;
            list1=list1->next;
        }
        while(list2){
            x=list2->val;
            curr->next= new ListNode(x);
            curr=curr->next;
            list2=list2->next;
        }
        return head.next;
    }
};